lib/errors/invalid-client-error.ts

```diff
@@ -0,0 +1,16 @@
+import { OAuthError } from './oauth-error';
+
+/**
+ * Constructor.
+ *
+ * "Client authentication failed (e.g., unknown client, no client
+ * authentication included, or unsupported authentication method)"
+ *
+ * @see https://tools.ietf.org/html/rfc6749#section-5.2
+ */
+
+export class InvalidClientError extends OAuthError {
+  constructor(message?: string | Error, properties?: any) {
+    super(message, { code: 400, name: 'invalid_client', ...properties });
```

Comment:
The status code should be `401 (Unauthorized)`

Reply:
https://tools.ietf.org/html/rfc6749#section-5.2
5.2.  Error Response

   The authorization server responds with an HTTP 400 (Bad Request)......
....
invalid_client
...
unauthorized_client
...